ticketutil/ticket.py

```diff
@@ -67,19 +105,21 @@ def _create_requests_session(self):
             r.raise_for_status()
             logging.debug("Create requests session: Status Code: {0}".format(r.status_code))
             logging.info("Successfully authenticated to {0}".format(self.ticketing_tool))
+            return s
```

Comment:
I would propose creating attribute here instead of returning it. But maybe I am missing some use case. Then I would propose raising error in try block immediately after cleanup. Instead:
```python
        try:
            r = s.get(self.auth_url)
            r.raise_for_status()
            logging.debug("Create requests session: Status Code: {0}".format(r.status_code))
            logging.info("Successfully authenticated to {0}".format(self.ticketing_tool))
            return s
        except requests.RequestException as e:
            logging.error("Error authenticating to {0}.".format(self.auth_url))
            logging.error(e.args[0])
            s.close()
```
I would propose:
```python
        try:
            r = s.get(self.auth_url)
            r.raise_for_status()
        except requests.RequestException as e:
            logging.error("Error authenticating to {0}.".format(self.auth_url))
            if e.args:
                logging.error(e.args[0])
            s.close()
            raise

        logging.debug("Create requests session: Status Code: {0}".format(r.status_code))
        logging.info("Successfully authenticated to {0}".format(self.ticketing_tool))
        self.s = s
```
and to remove `self.s = ` in `__init__`. I prefer to keep logical gears in single place whenever possible. But I understand the nice thing that you see explicitly new attribute being created in `__init__`. IMHO to reraise immediately is better. But I see returning new session and adding it as an attribute in `__init__` has its merits too.

But again maybe I am missing some reason for this...